### PR TITLE
solved issue #7

### LINKED ISSUE #7: resolve-bin should not throw when package.json does not contains a "main" field.
https://github.com/thlorenz/resolve-bin/blob/3e9b84e04b690cb27b0e666cc01d9b7d8711d608/index.js#L25

`resolve-bin` leverages `require.resolve` to resolve any bin package in the node_modules and then read package.json for a "bin" field.

However, when package.json does not contain a `"main"` field, and the package does not provide `index.js`, `require.resolve` will throw according to Node.js [Module Resolution Algorithm](https://nodejs.org/api/modules.html#modules_all_together).

As a bin package, it is somehow popular that the package author does not expose a `"main"` field, examples are `@babel/cli` and `@vue/cli`.

Instead of changing the file layout or metadata of these packages, I propose we resolve against `package.json` instead:
```js
require.resolve(name + "/package.json")
```
so it can successfully resolve for these packages as `npm` does.

Additional Context: https://github.com/babel/babel/pull/10719

### PR DESCRIPTION
**Pull request by bingtimren**
This PR fixes #7 

With this PR resolve-bin now works with packages that does not has "main" field in "package.json". One example is "open-cli".

Also added tests.

Please accept this PR and publish a new release. Thanks

### CODE REVIEW

File: examples/open-cli.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var resolveBin = require('../');
+
+// package.json:
+//  "bin": {
+//    "mocha": "./bin/mocha",
+//    "_mocha": "./bin/_mocha"
+//  },
```
Comment (bendrucker) [inline examples/open-cli.js:9]: Copy pasted?

**Review by bendrucker — CHANGES_REQUESTED** [review]
Nice! If you can fix the copied bit in the example that refers to Mocha, happy to get this merged and released as a patch.

File: examples/open-cli.js (hunk 1 above)
Comment (bingtimren) [reply examples/open-cli.js]: Woops, copy pasted indeed. Fixed.

**Comment by bingtimren** [comment]
Found another example "dependency-cruise", the "package.json" cannot be resolved. Therefore changed the code to first attempt "require.resolve", which is identical with the current behaviour, and only attempts to resolve "package.json" if it fails.

**Comment by bendrucker** [comment]
Done!

**Pull request merged**